ext/dom/php_dom.c

```diff
@@ -1028,6 +1038,26 @@ zend_object *dom_objects_new(zend_class_entry *class_type)
 }
 /* }}} */
 
+static zend_object *dom_objects_namespace_node_new(zend_class_entry *class_type)
+{
+	ZEND_ASSERT(class_type == dom_namespace_node_class_entry);
```

Comment:
Would this be an issue if someone sets a custom class for namespace nodes?

Otherwise, just asserting it is an instance of the namespace node class should be fine.

Reply:
Right now it's not possible to register a custom class for DOMNameSpaceNode, likely because of an oversight.
I'll get rid of the assert anyway though, because this might be a problem when support is added in the future. The assert only had the purpose of sanity checking that I was using the right object handlers, so it's safe to remove.